src/components/network-area-diagram-viewer/network-area-diagram-viewer.ts

```diff
@@ -272,27 +282,41 @@ export class NetworkAreaDiagramViewer {
         if (svg != null) {
             svg.style.cursor = 'grabbing';
         }
-        this.initialPosition = DiagramUtils.getPosition(this.selectedElement); // used for the offset
-        this.edgeAngles = new Map<string, number>();
+        if (!(event as MouseEvent).shiftKey) {
+            this.initialPosition = DiagramUtils.getPosition(
+                this.selectedElement
+            ); // used for the offset
+            this.edgeAngles = new Map<string, number>();
+        }
     }
 
     private drag(event: Event) {
         if (this.selectedElement) {
             event.preventDefault();
-            this.ctm = this.svgDraw?.node.getScreenCTM();
-            const mousePosition = this.getMousePosition(event as MouseEvent);
-            this.updateGraph(mousePosition);
-            this.initialPosition = DiagramUtils.getPosition(
-                this.selectedElement
-            );
+            if (!(event as MouseEvent).shiftKey) {
```

Comment:
you shouldn't look again at that parameter, but you could introduce a boolean shiftKeyOnMouseDown instead.

Reply:
done